drivers/adc/ad9250/ad9250.c

```diff
@@ -546,28 +511,18 @@ int32_t ad9250_offset_adj(struct ad9250_dev *dev,
  *		 Example: 1 - Disables the data output;
  *			  0 - Enables the data output(default).
  *
- * @return  Returns negative error code or the output disable state.
+ * @return Returns negative error code or 0 in case of success.
  *******************************************************************************/
 int32_t ad9250_output_disable(struct ad9250_dev *dev,
 			      int32_t en)
 {
-	uint32_t ret = 0;
-
-	if((en == 0) || (en == 1)) {
-		ret = ad9250_set_bits_to_reg(dev,
-					     AD9250_REG_OUT_MODE,
-					     en * AD9250_OUT_MODE_DISABLE,
-					     AD9250_OUT_MODE_DISABLE);
-	} else {
-		ret = ad9250_read(dev,
-				  AD9250_REG_OUT_MODE);
-		if(ret != SUCCESS) {
-			return ret;
-		}
-		return (ret & AD9250_OUT_MODE_DISABLE) != 0;
-	}
-
-	return ret;
+	if((en == 0) || (en == 1))
+		return ad9250_set_bits_to_reg(dev,
```

Comment:
Same as above, and I see this repeats below. I think this commit does a bit too much and could be split, it would be easier to track changes and avoid mixing them up.

Reply:
If i split the commit, adding the `register_data` parameter, i would have to modify all the read functions that are used here by adding an extra variable to each function in the driver and changing all headers to support also the data read.

All of this just to delete them in another commit.

I know it's a bit messy, but this driver was already broken.